### PR TITLE
feat: introduce Tigera Operator and Policy Only mode for EKS.

### LINKED ISSUE #38: Calico policy-only mode
We should add a package to this module to provide a Calico installation in policy-only mode. In this mode, Calico provides only policy capabilities while CNI is delegated to some other third-party component.

The main use case of this issue would be providing network policy capabilities in EKS: https://docs.aws.amazon.com/eks/latest/userguide/calico.html

### LINKED ISSUE #39: Add tigera operator package
It would be nice to have a new package in the networking module with the `calico-tigera-operator`.

This operator would make easier the installation of the networking and could also be a fix to have #38 as just a configuration option instead of having to maintain two different packages.

The migration from the manifests-based deployment to the operator-based one is pretty straightforward according to the official documentation:
https://projectcalico.docs.tigera.io/maintenance/operator-migration

### PR DESCRIPTION
**Pull request by ralgozino**
Introduce the Tigera Operator to the Networking module. Including:

- Operator Installation and configuration for on-prem installations.
- Policy-only mode for EKS clusters.

Tested:
- [x] Installation from scratch in an on-prem cluster
- [x] Upgrade from a manifests installation

Fixes #38
Fixes #39

### CODE REVIEW

File: katalog/tigera/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+> ⚠️ If you were using "**Infra**" nodes, you'll need to patch the `Install` resource with the right `NodeSelector` and `Tolerations`.
+> For example with the following Kustomize `patchesStrategicMerge`:
+>
+> ```yaml
+> ---
+> apiVersion: operator.tigera.io/v1
+> kind: Installation
+> metadata:
+>   name: default
```
Comment (lnovara) [inline katalog/tigera/README.md:44]: Is this CR namespaced?

**Review by lnovara — CHANGES_REQUESTED** [review]
Left some comments.

What is our mid/long-term plan for this module? Do we want to fully move to Tigera? What are the upstream project's recommendations about that?

File: katalog/tigera/README.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+>   controlPlaneNodeSelector:
+>     node.kubernetes.io/role: "infra"
+>   controlPlaneTolerations:
+>     - key: node.kubernetes.io/role
+>       value: infra
+>       operator: Equal
+>       effect: NoSchedule
+> ```
+
+The Operator moves Calico's pods from the `kube-system` namespace to the `calico-namespace`. As part of the `on-prem` variant, you are provided with the new resources (`Services` and `ServiceMonitors`) to keep the monitoring working in the new namespace.
```
[the remark this answers is not in the document]
Comment (ralgozino) [reply katalog/tigera/README.md:55]: nice catch!

File: katalog/tigera/README.md (hunk 1 above)
Comment (ralgozino) [reply katalog/tigera/README.md:44]: It is cluster-scoped. See line number 5464 of the CRD definition file `tigera-operator.yaml`

File: katalog/tigera/eks-policy-only/calico-crs.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+# license that can be found in the LICENSE file.
+
+---
+# Source: aws-calico/templates/crs/custom-resources.yaml
+# This section includes base Calico installation configuration.
+# For more information, see: https://docs.projectcalico.org/v3.17/reference/installation/api#operator.tigera.io/v1.Installation
+apiVersion: operator.tigera.io/v1
+kind: Installation
+metadata:
+  name: default
```
[the remark this answers is not in the document]
Comment (ralgozino) [reply katalog/tigera/eks-policy-only/calico-crs.yaml:12]: It is cluster-scoped. See line number 5464 of the CRD definition file `tigera-operator.yaml`

File: katalog/tigera/on-prem/custom-resources.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# For more information, see: https://projectcalico.docs.tigera.io/master/reference/installation/api#operator.tigera.io/v1.Installation
+apiVersion: operator.tigera.io/v1
+kind: Installation
+metadata:
+  name: default
+# we just let the Operator figure out the configuration
+spec:
+  variant: Calico
+  registry: registry.sighup.io/fury/
+  # Configures Calico networking.
```
[the remark this answers is not in the document]
Comment (ralgozino) [reply katalog/tigera/on-prem/custom-resources.yaml:16]: When [migrating from Calico installed by manifests](https://projectcalico.docs.tigera.io/maintenance/operator-migration), the Operator will detect the current configuration.
When a clean installation it should detect kubeadm's configuration, so "autodetect" the right CIDR*.

*I'm sure I read that on the docs for the manifests installation, I need to verify it for the operator.

[the remark this answers is not in the document]
Comment (ralgozino) [reply katalog/tigera/on-prem/custom-resources.yaml:16]: For the manifests installation [the docs](https://projectcalico.docs.tigera.io/getting-started/kubernetes/self-managed-onprem/onpremises) say the following:
image 

I could not find similar information in the docs for the operator, but I just did a local test with an "on-prem" cluster using our kubeadm-based installer, and the Operator installed as is provided in this PR autodetects the right POD and SVC CIDRs from the Kubeadm configuration:
image


When I tried the migration from manifests to the operator, the CIDRs were the same after switching to the operator.

I hope to have answered the question 😄

**Comment by ralgozino** [comment]
> Left some comments.
> 
> What is our mid/long-term plan for this module? Do we want to fully move to Tigera? What are the upstream project's recommendations about that?

The main idea behind introducing the operator is to provide the policy-only mode for EKS clusters, so we are not missing any pieces there. The on-prem deployment for the operator was "for free", I had to do it anyway for the policy-only mode.

Even though upstream still maintains the manifests installations they are pushing the Operator-based installation, and even say that the manifests installation is "not recommended", so we'll have to eventually migrate to the operator IMO.

> Calico can also be installed using raw manifests as an alternative to the operator. The manifests contain the necessary resources for installing Calico on each node in your Kubernetes cluster. Using manifests is not recommended as they cannot automatically manage the lifecycle of the Calico as the operator does. However, manifests may be useful for clusters that require highly specific modifications to the underlying Kubernetes resources.
> From: https://projectcalico.docs.tigera.io/getting-started/kubernetes/self-managed-onprem/onpremises

I'd like to introduce the operator in this release and after some testing start migrating over (unless we find any blockers), this is the reason why I also did not update the main readme changing the instructions to the operator.

**Review by nutellinoit — APPROVED** [review]
Overall LGTM 🎉 

Only some micro changes

File: katalog/tigera/README.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+# Tigera Package
+
+The Tigera package provides the Tigera Operator and some ready to go configurations to enable Networking capabilities for a Kubernetes cluster.
```
[the remark this answers is not in the document]
Comment (ralgozino) [reply katalog/tigera/README.md:3]: The `and` is actually OK IMO

[the remark this answers is not in the document]
Comment (nutellinoit) [reply katalog/tigera/README.md:3]: Yes, you are right :D

**Review by lnovara — APPROVED** [review]
SBAM! :shipit:

**Pull request merged**